packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -404,6 +387,49 @@ export class StateMachine extends StateMachineBase {
   public metricTime(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
     return this.metric('ExecutionTime', props);
   }
+
+  private buildLoggingConfiguration(props: StateMachineProps) {
+    if (props.logs === undefined) {
+      return undefined;
+    }
+
+    // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
+    this.addToRolePolicy(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        'logs:CreateLogDelivery',
+        'logs:GetLogDelivery',
+        'logs:UpdateLogDelivery',
+        'logs:DeleteLogDelivery',
+        'logs:ListLogDeliveries',
+        'logs:PutResourcePolicy',
+        'logs:DescribeResourcePolicies',
+        'logs:DescribeLogGroups',
+      ],
+      resources: ['*'],
+    }));
+
+    return {
+      destinations: [{ cloudWatchLogsLogGroup: { logGroupArn: props.logs.destination.logGroupArn } }],
+      includeExecutionData: props.logs.includeExecutionData,
+      level: props.logs.level || 'ERROR',
+    };
+  }
+
+  private buildTracingConfiguration(props: StateMachineProps) {
+    if (!props.tracingEnabled) {
+      return undefined;
+    }
+
+    this.addToRolePolicy(new iam.PolicyStatement({
```

Comment:
also worth linking to documentation as to why these cannot be scoped to resource level - i.e. point to https://docs.aws.amazon.com/xray/latest/devguide/security_iam_id-based-policy-examples.html#xray-permissions-resources